internal/docstore/docstore.go

```diff
@@ -179,21 +195,44 @@ func (l *ActionList) Update(doc Document, mods Mods) *ActionList {
 // See ActionList.Update.
 type Mods map[FieldPath]interface{}
 
-// Do executes the action list. If all the actions executed successfully, Do returns
-// (number of actions, nil). If any failed, Do returns the number of successful
-// actions and an error. In general there is no way to know which actions succeeded,
-// but the error will contain as much information as possible about the failures.
-func (l *ActionList) Do(ctx context.Context) (int, error) {
+// An ActionListError is returned by ActionList.Do. It contains all the errors
+// encountered while executing the ActionList, and the positions of the corresponding
+// actions.
+type ActionListError = driver.ActionListError
+
+// Do executes the action list.
+// In ordered mode (when the Unordered method was not called on the list), execution
+// will stop after the first action that fails.
+// In unordered mode, all the actions will be executed. In either mode, as a special
```

Comment:
Consider moving this "none of the actions" thing to be first? E.g., 

```
// Do executes the action list.
//
// If any of the actions are invalid (for example, ...), it returns an error, and no actions are executed.
//
// If ordered mode...
//
// In unordered mode...
```

Reply:
I still think this (the "special case") should be moved up above, or at least not be part of the "unordered mode" paragraph (since it applies to both modes).

Reply:
Done.